pipeline/pipelines/code_generation_pipeline.py

```diff
@@ -71,4 +73,23 @@ def additional_tasks_for_remote_execution(self, **kwargs):
                                          inject=kwargs)]
 
     def validate_kwargs(self, **kwargs):
-        _validate_codegen_kwargs([], **kwargs)
+        _validate_codegen_kwargs(self.task_factory.get_validate_kwargs(),
+                                 **kwargs)
+
+
+class TaskFactoryBase(object):
```

Comment:
optional: The Python standard library provides a way to do this via the abc metaclass, but afaik it's essentially equivalent to what you've done here anyway.